metricflow/query/query_parser.py

```diff
@@ -476,7 +483,7 @@ def _parse_and_validate_query(
             order_by_specs=order_by_specs,
             output_column_name_overrides=tuple(output_column_name_overrides),
             time_range_constraint=time_constraint,
-            where_constraint=spec_where_constraint,
+            where_constraint=resolved_where_filter,
```

Comment:
I can't tell if all of the changes here are correct. Clearly, _parse_and_validate_query is due for some tidying. But not now.